content/en/blog/_posts/2025-10-25-new-cgroup-v1-to-v2-conversion-formula.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: blog
+title: 'New conversion from cgroup v1 CPU shares to v2 CPU weight'
+date: 2025-10-25T05:00:00-08:00
+draft: true
+math: true
+slug: new-cgroup-v1-to-v2-cpu-conversion-formula
+author: >
+   [Itamar Holder](https://github.com/iholder101) (Red Hat)
+---
+
+I'm excited to announce the implementation of an improved conversion formula 
+from cgroup v1 CPU shares to cgroup v2 CPU weight. This enhancement addresses 
+critical issues with CPU priority allocation for Kubernetes workloads when 
+running on systems with cgroup v2.
+
+## Background
+
+Kubernetes was originally designed with cgroup v1 in mind, where CPU shares 
+were defined simply by assigning the container's CPU requests in millicpu 
+form. 
+
+For example, a container requesting 1 CPU (1024m) would get \(cpu.shares = 1024\).
+
+After a while, cgroup v1 was stared being replaced by its successor, 
+cgroup v2. In cgroup v2, the concept of CPU shares (which ranges from 2 to 
+262144, or from \(2^1\) to \(2^18\)) was replaced with CPU weight (which ranges from 
+\([1, 10000]\), or \(10^0\) to \(10^4\).
+
+With the transition to cgroup v2, 
+[KEP-2254](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2) 
+introduced a conversion formula to map cgroup v1 CPU shares to cgroup v2 CPU 
+weight. The conversion formula was defined as:
+
+```math
+cpu.weight = (1 + ((cpu.shares - 2) * 9999) / 262142)
+```
+
+This formula linearly maps between \([2^1 - 2^18]\) to \([10^0 - 10^4]\).
+![2025-10-25-new-cgroup-v1-to-v2-conversion-formula-linear-conversion.png](2025-10-25-new-cgroup-v1-to-v2-conversion-formula-linear-conversion.png)
```

Comment:
Not shown
https://deploy-preview-52793--kubernetes-io-main-staging.netlify.app/blog/2025/10/25/new-cgroup-v1-to-v2-cpu-conversion-formula/

Reply:
Turns out the post file needs to be named `index.md` :man_facepalming: 
Should be fine now